gpflow/kernels/periodic.py

```diff
@@ -9,6 +9,9 @@
 from .stationaries import Stationary
 
 
+_ACTIVE_DIMS_ON_BASE_KERNEL = 'active_dims should be specified through the base kernel'
```

Comment:
What is that? And why is that a default value for active_dims?
actives_dims is either None, a slice or an array, but not a string 100% :)

Reply:
It's a sentinel value to detect whether someone passes an `active_dims` argument to the `Periodic` constructor. And makes it obvious when looking at e.g. `Periodic?` in IPython that you shouldn't actually pass anything in there. The issue is that currently `None` doesn't mean "active_dims hasn't been passed in" but "apply to all dimensions", so passing `active_dims=None` to the periodic kernel when the base kernel has active_dims set to something else should throw an error.

Reply:
The sentinel value cannot violate type restrictions, the active dim is an **optional** argument, and can be only a **slice** or an **array**. Here you can say that periodic class accepts only none value and doesn't tolerate anything else; state it in the docstring and raise an exception in case of when it is not none.

Reply:
I don't understand the argument with `IPython`.

Reply:
```
In [2]: gpflow.kernels.Periodic?                                                                                
Init signature:
gpflow.kernels.Periodic(
    base: gpflow.kernels.stationaries.Stationary,
    period: Union[float, List[float]] = 1.0,
    active_dims='active_dims should be specified through the base kernel',
)
```

Reply:
Alternatively we can use **kwargs and check whether they contain `"active_dims"` (and then check consistency with `base.active_dims`)?

Reply:
> Here you can say that periodic class accepts only none value and doesn't tolerate anything else; state it in the docstring and raise an exception in case of when it is not none.

My point was that because `None` stands for "all input dimensions" we *cannot* use it to indicate that this is the only accepted value.

Reply:
Let's not allow `IPython` to rule our coding standards. The string is unacceptable. It should accept only `None` value. What we need to do: 1. change type annotation to emphasise it, 2. make docstring clear about it. 

Sorry, but IPython has nothing to do with it.

Reply:
changed to **kwargs and corresponding check - happy?